### ISSUE TITLE
TS2322: Type 'string | number | any[] | { value: any; symbol: any; exponent: number; unit: string; }' is not assignable to type 'ReactNode'.

### ISSUE THREAD

**Issue by greenpau** [avoidwork/filesize.js#167]
After the upgrade to React 18, I started getting the following Typescript errors:

```
ERROR in src/components/CreateJob/ReviewStep.tsx:102:47
TS2322: Type 'string | number | any[] | { value: any; symbol: any; exponent: number; unit: string; }' is not assignable to type 'ReactNode'.
  Type '{ value: any; symbol: any; exponent: number; unit: string; }' is not assignable to type 'ReactNode'.
    100 |               header: 'File Size',
    101 |               cell: (item) => {
  > 102 |                 return <Box textAlign="right">{filesize(item.size, { base: 2, standard: 'jedec' }) || '-'}</Box>;
        |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    103 |               },
    104 |             },
    105 |           ]}

ERROR in src/components/FileUpload/FileUploadTable.tsx:227:39
TS2322: Type 'string | number | any[] | { value: any; symbol: any; exponent: number; unit: string; }' is not assignable to type 'ReactNode'.
    225 |       header: 'Size',
    226 |       cell: (item) => {
  > 227 |         return <Box textAlign="right">{filesize(item.size, { base: 2, standard: 'jedec' }) || '-'}</Box>;
        |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    228 |       },
    229 |       sortingField: 'size',
    230 |     },

ERROR in src/components/FileUpload/StagedFileList.tsx:48:43
TS2322: Type 'string | number | any[] | { value: any; symbol: any; exponent: number; unit: string; }' is not assignable to type 'ReactNode'.
  Type '{ value: any; symbol: any; exponent: number; unit: string; }' is not assignable to type 'ReactNode'.
    46 |           header: 'File Size',
    47 |           cell: (item) => {
  > 48 |             return <Box textAlign="right">{filesize(item.file.size, { base: 2, standard: 'jedec' }) || '-'}</Box>;
       |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    49 |           },
    50 |         },
    51 |         {
```

**Comment by avoidwork** [comment]
That file is generated.

**Comment by greenpau** [comment]
> That file is generated; tsc bug?

Yes. There is some difference between React 17 and React 18 handling of the `filesize` module.

**Comment by avoidwork** [comment]
ok, i'll just swap that in place.

**Comment by avoidwork** [comment]
@greenpau are you using @types/react 18.0.0? from what i'm reading this has nothing to do with filesize and simply relates to the optional object output.

**Comment by felixhoffmnn** [comment]
I ran into the same problem. As far as I understand, this is only kind of related to `@types/react`. The main problem is that the return type from the `filesize()` function can be of the following types:

```typescript
string | number | any[] | {
    value: any;
    symbol: any;
    exponent: number;
    unit: string;
};
```

If you want to use the result of the `filesize()` function inside a JSX element, the expected type is a string, integer, or iterable, which does not quite match the defined types. 

I was able to fix the type error by just parsing the result to a string (like `filesize().toString()`).
